algolia/transport/transport.go

```diff
@@ -26,7 +26,7 @@ const version = "3.30.0"
 type Transport struct {
 	requester     Requester
 	retryStrategy *RetryStrategy
-	headers       map[string]string
+	headers       http.Header
```

Comment:
It's a breaking change no? We could add another field taking precedence if provided.

Reply:
fwiw, in the past we used to accept BC on bug fixes.

Reply:
> It's a breaking change no? We could add another field taking precedence if provided.

So before, if we were playing with difference case, we were appending values together while now it overrides following this precedence:
- Request header
- Extra client header
- Default client header